test/plugins/slo/slo_test.go

```diff
@@ -2,23 +2,93 @@ package plugins_test
 
 import (
 	"context"
+	"fmt"
 	"time"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
 	managementv1 "github.com/rancher/opni/pkg/apis/management/v1"
+	"github.com/rancher/opni/pkg/slo/query"
 	"github.com/rancher/opni/pkg/slo/shared"
 	"github.com/rancher/opni/pkg/test"
+	"github.com/rancher/opni/plugins/cortex/pkg/apis/cortexadmin"
 	apis "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	"github.com/rancher/opni/plugins/slo/pkg/slo"
 	"google.golang.org/protobuf/types/known/durationpb"
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
+func expectSLOGroupToExist(adminClient cortexadmin.CortexAdminClient, ctx context.Context, tenant string, groupName string) error {
+	var anyError error
+	if err := expectRuleGroupToExist(adminClient, ctx, tenant, groupName+slo.RecordingRuleSuffix); err != nil {
+		anyError = err
+	}
+	if err := expectRuleGroupToExist(adminClient, ctx, tenant, groupName+slo.MetadataRuleSuffix); err != nil {
+		anyError = err
+	}
+	if err := expectRuleGroupToExist(adminClient, ctx, tenant, groupName+slo.AlertRuleSuffix); err != nil {
+		anyError = err
+	}
+	return anyError
+}
+
+func expectSLOGroupToNotExist(adminClient cortexadmin.CortexAdminClient, ctx context.Context, tenant string, groupName string) error {
+	var anyError error
+	if err := expectRuleGroupToNotExist(adminClient, ctx, tenant, groupName+slo.RecordingRuleSuffix); err != nil {
+		anyError = err
+	}
+
+	if err := expectRuleGroupToNotExist(adminClient, ctx, tenant, groupName+slo.MetadataRuleSuffix); err != nil {
+		anyError = err
+	}
+
+	if err := expectRuleGroupToNotExist(adminClient, ctx, tenant, groupName+slo.AlertRuleSuffix); err != nil {
+		anyError = err
+	}
+	return anyError
+}
+
+func expectRuleGroupToExist(adminClient cortexadmin.CortexAdminClient, ctx context.Context, tenant string, groupName string) error {
+	for i := 0; i < 10; i++ {
```

Comment:
these helper functions are good, but you should implement them using `Eventually()`.

Reply:
Refactored them using `sync.WaitGroup` and `go func(){}` instead of `Eventually()` since I had trouble refactoring them correctly to an Eventually matcher